pkg/odo/cli/describe/component.go

```diff
@@ -306,9 +308,27 @@ func (o *ComponentOptions) describeDevfileComponent(ctx context.Context) (result
 		// Display RunningOn field only if the feature is enabled
 		cmp.RunningOn = nil
 	}
+	updateWithRemoteSourceLocation(&cmp)
 	return cmp, devfileObj, err
 }
 
+func updateWithRemoteSourceLocation(cmp *api.Component) {
+	components, err := cmp.DevfileData.Devfile.GetComponents(common.DevfileOptions{
+		ComponentOptions: common.ComponentOptions{ComponentType: v1alpha2.ContainerComponentType},
+	})
+	if err != nil {
+		return
+	}
+	for _, comp := range components {
+		if *comp.Container.MountSources {
+			if comp.Container.SourceMapping == "" {
+				comp.Container.SourceMapping = generator.DevfileSourceVolumeMount
+				_ = cmp.DevfileData.Devfile.UpdateComponent(comp)
```

Comment:
Could we also `klog` the error if any instead of ignoring it? You never know! :)